src/plugins/data/public/field_formats_provider/field_formats.ts

```diff
@@ -0,0 +1,224 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { forOwn, isFunction, memoize } from 'lodash';
+import { UiSettingsClientContract } from 'kibana/public';
+import { FIELD_FORMATS_IDS } from './types';
+import { ES_FIELD_TYPES, KBN_FIELD_TYPES, FieldFormat } from '../../common';
+
+interface FieldType {
+  id: FIELD_FORMATS_IDS;
+  params: Record<string, any>;
+  es?: boolean;
+}
+
+export class FieldFormatRegisty {
+  private fieldFormats: Map<FIELD_FORMATS_IDS, FieldFormat['constructor']>;
+  private uiSettings!: UiSettingsClientContract;
+  private defaultMap: { [key in ES_FIELD_TYPES | KBN_FIELD_TYPES]?: FieldType } & {
```

Comment:
Could you please show me cases where key is a ES_FIELD_TYPE and where it's a KBN_FIELD_TYPE ? I just want to be sure that it's a right type

Reply:
Make as a `string`